app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/event_screen/EventEditors.kt

```diff
@@ -0,0 +1,268 @@
+package com.realityexpander.tasky.agenda_feature.presentation.event_screen
+
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material.icons.filled.Error
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.DialogProperties
+import com.realityexpander.tasky.R
+import com.realityexpander.tasky.agenda_feature.presentation.common.components.EditTextModal
+import com.realityexpander.tasky.agenda_feature.presentation.common.util.toZonedDateTime
+import com.realityexpander.tasky.agenda_feature.presentation.event_screen.EventScreenEvent.*
+import com.realityexpander.tasky.core.presentation.common.modifiers.DP
+import com.realityexpander.tasky.core.presentation.common.modifiers.mediumHeight
+import com.realityexpander.tasky.core.presentation.common.modifiers.smallHeight
+import com.realityexpander.tasky.core.presentation.common.modifiers.tinyHeight
+import com.vanpra.composematerialdialogs.MaterialDialog
+import com.vanpra.composematerialdialogs.datetime.date.datepicker
+import com.vanpra.composematerialdialogs.datetime.time.timepicker
+import com.vanpra.composematerialdialogs.rememberMaterialDialogState
+import java.time.LocalDateTime
+
+@Composable
+fun EventPropertyEditors(
+    editMode: EditMode,
+    state: EventScreenState,
+    onAction: (EventScreenEvent) -> Unit,
+) {
+    when (editMode) {
+        is EditMode.ChooseTitleText,
+        is EditMode.ChooseDescriptionText -> {
+            editMode as EditMode.TextPayload
+
+            EditTextModal(
+                text = editMode.text,
+                title = editMode.dialogTitle.get,
+                editTextStyle =
+                if (editMode is EditMode.ChooseTitleText)
+                    MaterialTheme.typography.h2  // cant access from non-compose function, make a wrapper?
+                else
+                    MaterialTheme.typography.body1, // cant access from non-compose function, make a wrapper?
```

Comment:
Why does this need to be accessed from non-compose code?

Reply:
I was able to make it work from the call site, but the MaterialTheme can only be accessed from the @Composable.